src/main/java/com/atlassian/db/replica/internal/ReplicaConnectionProvider.java

```diff
@@ -57,14 +57,7 @@ public int getTransactionIsolation() throws SQLException {
     }
 
     public void setAutoCommit(Boolean autoCommit) throws SQLException {
-        final boolean autoCommitBefore = getAutoCommit();
-        if (autoCommitBefore != autoCommit) {
-            preCommit(autoCommitBefore);
-        }
         parameters.setAutoCommit(state::getConnection, autoCommit);
-        if (autoCommitBefore != getAutoCommit()) {
-            recordCommit(autoCommitBefore);
```

Comment:
Why did `ReplicaConsistency` need to know about `setAutoCommit`? Can this call cause an immediate commit?
I thought that the next `executeUpdate` could do that implicitly, but nothing happens in `setAutoCommit` itself.

Reply:
I added the integration tests. One fails when introduced initially and goes green after removing auto-commit handling.

Reply:
> Why did ReplicaConsistency need to know about setAutoCommit? Can this call cause an immediate commit?

Yes. It's a part of the connection API. 
```
   * <B>NOTE:</B>  If this method is called during a transaction and the
     * auto-commit mode is changed, the transaction is committed.  If
     * <code>setAutoCommit</code> is called and the auto-commit mode is
     * not changed, the call is a no-op.
```